[Chen Yiyang] iP

Chen-Yiyang opened the pull request:
# Yiyang-Bot
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Yiyang-Bot frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😅 

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

-----
If you Java programmer, you can use it to practice Java too. Here's the `main` method:
``` Java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Chen-Yiyang committed:
Basic Greet, Echo and Exit displays

Chen-Yiyang committed:
Implement add and list commands. Add in a separate Task class.

Chen-Yiyang committed:
Level-3 Implement Mark-as-Done

Chen-Yiyang committed:
Level 4 Implement Todo, Event and Deadline using inheritance.

Chen-Yiyang committed:
A-TextUiTesting Add in automated testing.

Chen-Yiyang committed:
Level-5 Exception handling.

Chen-Yiyang committed:
Level-6 Implement delete.

Chen-Yiyang committed:
Implement Storage load

Chen-Yiyang committed:
Implement Storage load

Chen-Yiyang committed:
Modift .gitignore to exclude data storage file

Chen-Yiyang committed:
Redefine saved item format in local storage

Chen-Yiyang committed:
Implement LocalDate for Deadline and Event

Chen-Yiyang committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Implement LocalDate for Deadline and Event

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Chen-Yiyang committed:
Fix bugs with parsing stored data

Chen-Yiyang committed:
Restructure code to follow OOP with TaskList and Ui classes

Chen-Yiyang committed:
Improve OOP with Command and Parser

Chen-Yiyang committed:
Divide classes into packages

Chen-Yiyang committed:
Modify runtest

Chen-Yiyang committed:
Use JUnit for testing

Chen-Yiyang committed:
Autogenerate manifest file for building jar

Chen-Yiyang committed:
Add JavaDocs

Chen-Yiyang committed:
Modify code to follow the coding standard

Chen-Yiyang committed:
Implement find task

Chen-Yiyang committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Modify code to follow the coding standard

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/Event.java

Chen-Yiyang committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Implement find task

Chen-Yiyang committed:
Merge branch 'add-gradle-support' of https://github.com/Chen-Yiyang/ip into branch-A-Gradle

* 'add-gradle-support' of https://github.com/Chen-Yiyang/ip:
  Add Gradle support

Chen-Yiyang committed:
Merge branch 'branch-A-Gradle'

* branch-A-Gradle:
  Add Gradle support

Chen-Yiyang committed:
Setup GUI layout using javafx

Chen-Yiyang committed:
Implement simple dialogue box for GUI

Chen-Yiyang committed:
Implement javafx controller using fxml

Chen-Yiyang committed:
Incorporate bot logic into GUI

Chen-Yiyang committed:
Improve UI and remove unnecessary code

hmanhduc2k commented on src/main/java/duke/Parser.java:
```diff
@@ -11,6 +11,7 @@
     private static final String CMD_DDL = "deadline";
     private static final String CMD_EVENT = "event";
     private static final String CMD_DEL = "delete";
+    private static final String CMD_FIND = "find";
```
Do consider making your variable declaration clearer, in a sense that people who are reading your code may not understand what CMD is referring to, or what DDL is referring to. For example, COMMAND_DEADLINE or something like that

hmanhduc2k commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,24 @@
+        String keyword = fullCommand.substring(4).strip();
+        taskList.displayFind(keyword);
+    }
+}
```
Consider adding JavaDoc comments to your code to make people understand what the code is trying to do. If it is finding, then explain those with JavaDoc header.

hmanhduc2k commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,24 @@
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        String keyword = fullCommand.substring(4).strip();
+        taskList.displayFind(keyword);
```
Try to avoid magic number in this case, people reading "4" might not understand why. I believe that a split command based on spacebar notation will make people understand that you are trying to separate the first word from the rest of the command line

hmanhduc2k commented on src/main/java/duke/command/FindCommand.java:
```diff
@@ -0,0 +1,24 @@
+import duke.TaskList;
+import duke.Ui;
+
+public class FindCommand extends Command{
+    private final String fullCommand;
+
```
Add space after Command to match coding standard?

hmanhduc2k reviewed (commented):
Overall, your code is really good, you have no problem in terms of standard violation. There are some minor things I just wanted to point out for you to consider, in making your code more readable and match the standard given in CS2103T

TTraveller7 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    /**
+     * parses a command and generates a new Deadline instance.
```
```suggestion
     * Parses a command and generates a new Deadline instance.
```

TTraveller7 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,104 @@
+
+        //if (isExit)
+
+        return new ResponseMessage(String.join("\n", linesOfOutput), isExit);
```
Is this line redundant?

TTraveller7 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    /**
+     * parses a command and generates a new Event instance.
```
```suggestion
     * Parses a command and generates a new Event instance.
```

TTraveller7 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,53 @@
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+
+public class AddCommand extends Command{
```
```suggestion
public class AddCommand extends Command {
```

TTraveller7 reviewed (commented):
Overall, code was well written and easy to understand! Some coding standard violations found, more test methods on invalid inputs could be useful.

Chen-Yiyang committed:
Update build.gradle to run Launcher class

Chen-Yiyang committed:
Add assertions

Use `assert` statements to add assertions in the projectly. Assertions are added mainly for input-related operations.

Chen-Yiyang committed:
Merge pull request #1 from Chen-Yiyang/branch-A-Assertions

Add assertions

Chen-Yiyang committed:
Remove unnecessary code comments

Follow the idea that code comments should only be included
if they answer the question of WHAT or WHY, but not HOW.
Unnecessary code comments are therefore removed to enhance code quality.

Chen-Yiyang committed:
Remove dead code to enhance code quality

Chen-Yiyang committed:
Merge pull request #2 from Chen-Yiyang/branch-A-CodeQuality

Branch a code quality

Chen-Yiyang committed:
Use Stream for Task methods

Chen-Yiyang committed:
Merge pull request #3 from Chen-Yiyang/branch-A-Streams

Use Stream for Task methods

Chen-Yiyang committed:
Fix bug with Stream for finding items

Chen-Yiyang committed:
Merge pull request #4 from Chen-Yiyang/branch-A-Streams

Fix bug with Stream for finding items

Chen-Yiyang committed:
Implement archive command

Chen-Yiyang committed:
Implement storage for archived items

Chen-Yiyang committed:
Improve GUI

Chen-Yiyang committed:
Handle exception messages in JavaFX display

Chen-Yiyang committed:
Include screenshot and user guide
